flocker/node/functional/test_deploy.py

```diff
@@ -35,8 +35,10 @@ def test_restart(self):
         """
         name = random_name()
         docker_client = DockerClient()
-        deployer = Deployer(create_volume_service(self), docker_client,
-                            make_memory_network())
+        deployer = P2PNodeDeployer(
+            u"localhost",
```

Comment:
Is it? We've seen issues with external IP detection, for example, and I would argue that hostnames are more stable than IPs. Which could be completely wrong. Anyway seems like something to discuss separately.